spec/prawn/font_spec.rb

```diff
@@ -111,11 +111,123 @@
     end
   end
 
+  describe '#font' do
+    it 'allows setting of size directly when font is created' do
+      pdf.font 'Courier', size: 16
+      expect(pdf.font_size).to eq(16)
+    end
+
+    it 'allows temporary setting of a new font using a transaction' do
+      pdf.font 'Helvetica', size: 12
+
+      pdf.font 'Courier', size: 16, style: :bold do
+        expect(pdf.font_family).to eq('Courier')
+        expect(pdf.font_size).to eq(16)
+        expect(pdf.font_style).to eq(:bold)
+      end
+
+      expect(pdf.font.name).to eq('Helvetica')
+      expect(pdf.font_size).to eq(12)
+      expect(pdf.font_style).to eq(:normal)
+    end
+
+    it 'masks font size when using a transacation' do
+      pdf.font 'Courier', size: 16 do
+        expect(pdf.font_size).to eq(16)
+      end
+
+      pdf.font 'Times-Roman'
+      pdf.font 'Courier'
+
+      expect(pdf.font_size).to eq(12)
+    end
+  end
+
   describe '#font_size' do
+    it 'returns default font size' do
+      expect(pdf.font_size).to eq(12)
+    end
+
     it 'allows setting font size in DSL style' do
       pdf.font_size 20
       expect(pdf.font_size).to eq(20)
     end
+
+    it 'allows setting font size in DSL style using a transaction' do
+      pdf.font_size 20 do
+        expect(pdf.font_size).to eq(20)
+      end
+
+      expect(pdf.font_size).to eq(12)
+    end
+
+    it 'allows setting font size as assignment' do
+      pdf.font_size = 20
+      expect(pdf.font_size).to eq(20)
+    end
+  end
+
+  describe '#font_style' do
+    it 'returns default font style' do
+      expect(pdf.font_style).to eq(:normal)
+    end
+
+    it 'allows setting font style in DSL style' do
+      pdf.font_style :bold
+      expect(pdf.font_style).to eq(:bold)
+      expect(pdf.font.name).to eq('Helvetica-Bold')
+    end
+
+    it 'allows setting font style in DSL style using a transaction' do
+      pdf.font_style :bold do
+        expect(pdf.font_style).to eq(:bold)
+        expect(pdf.font.name).to eq('Helvetica-Bold')
+      end
+
+      expect(pdf.font_style).to eq(:normal)
+      expect(pdf.font.name).to eq('Helvetica')
+    end
+
+    it 'allows setting font style for a TTF font' do
+      pdf.font_families.update(
+        'DejaVu Sans' => {
```

Comment:
Thanks for the test case. However, I think to really test this out we need another entry where the key is different from the font family name, e.g. using 'deja' as key.

I don't think that the current implementation would work since the look-up key would be the family name from the TTF.